lib/publiccloud/provider.pm

```diff
@@ -634,7 +634,7 @@ This method is called called after each test on failure or success.
 =cut
 sub cleanup {
     my ($self) = @_;
-    $self->terraform_destroy() unless script_run('test -d ' . TERRAFORM_DIR . '/.terraform');
+    $self->terraform_destroy() unless check_var('PUBLIC_CLOUD_SERVICE', 'EKS');
```

Comment:
I am not sure why you drop initial check , I think this would be correct version 
```suggestion
    $self->terraform_destroy() unless script_run('test -d ' . TERRAFORM_DIR . '/.terraform') && check_var('PUBLIC_CLOUD_SERVICE', 'EKS');
```

Reply:
The terraform dir check was introduced by the related PR so, I don't think is necessary, but yes, is more complete.

Reply:
Added

Reply:
@asmorodskyi Why conditioning the destroy to this `.terraform` directory? Maybe my lab for manual test is bad, but I always have this directory, before and after an apply or destroy command. In that case the destroy in cleanup is never done.

Reply:
you are right , this is basically what we trying to fix . sorry I was not careful enough . @ilausuch  please revert to your initial version sorry for confusing you